gradient/commands/datasets.py

```diff
@@ -676,3 +678,83 @@ def update_status():
                         for pre_signed in pre_signeds:
                             update_status()
                             pool.put(self._delete, url=pre_signed.url)
+
+
+class ImportDatasetCommand(BaseDatasetVersionsCommand):
+    def __init__(self, api_key):
+        print(api_key)
+        self.api_key = api_key
+
+    @classmethod
+    def create_secret(self, key, value, expires_in=86400):
+        print(self)
+        client = api_sdk.clients.SecretsClient(
+            api_key="ae2ea576ef51c046295bec0d004c67",
+            ps_client_name=CLI_PS_CLIENT_NAME,
+        )
+
+        response = client.ephemral(key, value, expires_in)
+        return response
+    @classmethod
+    def get_command_string(self, url):
+        return "go-getter {} /data/output".format(url)
+
+    @classmethod
+    def get_command(self, dataset_url, url_type, http_auth, access_key, secret_key):
+        if url_type == 'git' or url_type == 's3':
+            return self.get_command_string(dataset_url)
+        
+        if url_type == 'https':
+            if not http_auth:
+                return self.get_command_string(dataset_url)
+            u = urlparse(dataset_url)
+            http_auth_url = "https://${{HTTP_AUTH}}@{}".format(u.path)
+            return self.get_command_string(http_auth_url)
+
+        return ""
+    @classmethod
+    def get_env_vars(self, url_type, http_auth, access_key, secret_key):
+        if url_type == 'git':
+            return ""
+
+        if url_type == 's3':
+            access_key_secret = self.create_secret('AWS_ACCESS_KEY_ID', access_key)
+            secret_key_secret = self.create_secret('AWS_ACCESS_KEY_SECRET', secret_key)
+
+            access_key_value = "secret:ephemeral:{}".format(access_key_secret["AWS_ACCESS_KEY_ID"])
+            secret_key_value = "secret:ephemeral:{}".format(secret_key_secret["AWS_ACCESS_KEY_SECRET"])
+
+            return "{\"AWS_ACCESS_KEY_ID\":\"%s\", \"AWS_ACCESS_KEY_SECRET\":\"%s\"}" % (access_key_value, secret_key_value)
+
+        if url_type == 'https':
+            http_auth_secret = self.create_secret('HTTP_AUTH', http_auth)
+            return "{\"HTTP_AUTH\":\"%s\"}" % http_auth_secret
```

Comment:
can't you return a python dict and encode later?

Reply:
same as above